src/lib/Rating.svelte

```diff
@@ -0,0 +1,31 @@
+<script lang="ts">
+  export let value: number;
+  export let readonly: boolean = false;
+  // Name is only required when the component's value can change
+  export let name: string;
+
+  function inputChange(newValue: number) {
+    value = newValue;
+  }
+</script>
+
+<div class="rating rating-half block {$$restProps.class || ''}">
+  {#each Array(10) as _, n}
```

Comment:
```suggestion
  {#each Array(ratingLimit * 2) as _, n}
```
Kommer ikke på et bra navn. 10 er litt uklart.